src/app/shared/components/paginator/paginator.component.ts

```diff
@@ -10,14 +12,21 @@ import { Pageable } from '@app/core/api';
 })
 export class PaginatorComponent {
   @Input() length: number | undefined;
-  @Output() param: EventEmitter<Pageable> = new EventEmitter<Pageable>();
+  @Output() param: EventEmitter<void> = new EventEmitter<void>();
 
-  constructor() {}
+  public params$: Observable<Pageable>;
 
-  handlePageEvent(event: PageEvent) {
-    this.param.emit({
+  constructor(private router: Router, private route: ActivatedRoute) {
+    this.params$ = this.route.queryParams;
```

Comment:
Please move it to move on init. Keeping logic in constructor always cause problem.